sympy/matrices/tests/test_eigen.py

```diff
@@ -305,52 +299,47 @@ def test_bidiagonalize():
             comp = random.randint(-1000000000, 1000000000)
             value = real + comp * I
             test_values = test_values + [value]
-        M = Matrix(size, size, test_values)
-        N = ImmutableMatrix(M)
         # L     -> Lower Bidiagonalization
         # M     -> Mutable Matrix
         # N     -> Immutable Matrix
         # 0     -> Bidiagonalized form
         # 1,2,3 -> Bidiagonal_decomposition matrices
         # 4     -> Product of 1 2 3
-        N1, N2, N3 = N.bidiagonal_decomposition()
+        M = Matrix(size, size, test_values)
         M1, M2, M3 = M.bidiagonal_decomposition()
         M0 = M.bidiagonalize()
-        N0 = N.bidiagonalize()
-
-        N4 = N1 * N2 * N3
         M4 = M1 * M2 * M3
+        M0.simplify()
+        M2.simplify()
+        M4.simplify()
+        assert M == M4
+        assert M2 == M0
```

Comment:
All this code is duplicated above and below. You should just make a `test_matrix` function and use that.